commercetools-convenience/src/main/java/io/sphere/sdk/queries/QueryAllImpl.java

```diff
@@ -22,42 +24,140 @@ private QueryAllImpl(final QueryDsl<T, C> baseQuery, final long pageSize) {
         this.pageSize = pageSize;
     }
 
-    public CompletionStage<List<T>> run(final SphereClient client) {
+    /**
+     * Given a {@code callback} {@link Function}, this method applies this callback on each page of the results from
```

Comment:
Although this class isn't visible to our users, I like that you added javadoc here 👍

Reply:
463ce43f13